### DISCUSSION TITLE
Access model settings in Loopback 4

### DISCUSSION THREAD

**Discussion by condratbogdan** [loopbackio/loopback-next#7190] [Q&A]
In Loopback 3 i could access the models and their settings by using app.models['Logs'].settings.
How can i do this in Loopback 4?

**Comment by raymondfeng** [comment]
LB4 model classes are bound the `Application` context. You can use dependency injection or service locator to get a model class by name:

```ts
const settings = app.getSync('models.MyModel#.definition.settings');

Or

@inject('models.MyModel#definition.settings')
private settings: object;
```

**Comment by achrinza** [comment]
> const settings = app.getSync('models.MyModel#.definition.settings');

Would `app.get()` be preferred since it's non-blocking? AFAIK, `app.getSync()` exists only for use in a constructor, which can't be `async`-ed.

**Comment by raymondfeng** [comment]
> Would app.get() be preferred since it's non-blocking? AFAIK, app.getSync() exists only for use in a constructor, which can't be async-ed.

Not exactly. The difference between `getSync` and `get` is whether the binding can be resolved synchronously or asynchronously. If the binding resolution returns a promise by itself or its dependencies, `getSync` will throw an error. Please `await ctx.get(...)` works for both cases.

From performance perspective, `getSync` is more efficient than `get`. If the binding produces a value synchronously, using `get()` is the same as `getSync()` except the returned value is wrapped in a Promise.
